### PR TITLE
parser: XXXDatabaseStmt now use CIStr for DB name

### LINKED ISSUE #35622: AlterDatabaseStmt does not use CIStr
## Feature Request

**Is your feature request related to a problem? Please describe:**

AlterDatabaseStmt uses string to represent a database name.

**Describe the feature you'd like:**

use a case-sensitive string

**Describe alternatives you've considered:**

**Teachability, Documentation, Adoption, Migration Strategy:**

### PR DESCRIPTION
**Pull request by lance6716**
Signed-off-by: lance6716 <lance6716@gmail.com>

### What problem does this PR solve?

Issue Number: close #35622 

Problem Summary:

### What is changed and how it works?

as title

### Check List

Tests 

- pass existing tests
- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: planner/core/planbuilder.go [hunk 1]
```diff
@@ -4159,16 +4159,16 @@ func (b *PlanBuilder) buildDDL(ctx context.Context, node ast.DDLNode) (Plan, err
...
 		}
-		if v.Name == "" {
+		if v.Name.O == "" {
 			return nil, ErrNoDB
 		}
 		if b.ctx.GetSessionVars().User != nil {
 			authErr = ErrDBaccessDenied.GenWithStackByArgs("ALTER", b.ctx.GetSessionVars().User.AuthUsername,
 				b.ctx.GetSessionVars().User.AuthHostname, v.Name)
 		}
-		b.visitInfo = appendVisitInfo(b.visitInfo, mysql.AlterPriv, v.Name, "", "", authErr)
```
Comment (zimulala) [inline planner/core/planbuilder.go:4171]: I think using `v.Name.L` is better.

**Comment by lance6716** [comment]
CI passed, PTAL @zimulala @Defined2014

**Comment by lance6716** [comment]
PTAL @wjhuang2016 @tangenta

**Review by Defined2014 — APPROVED** [review]

**Review by tangenta — APPROVED** [review]

**Pull request merged**